common/utils.go

```diff
@@ -52,3 +52,35 @@ func URLEncode(s string) string {
 
 	return res
 }
+
+type pathBuilder struct {
+	sb                    strings.Builder
+	enableNameFieldEscape bool
+}
+
+func NewPathBuilder() *pathBuilder {
+	return &pathBuilder{}
+}
+
+func (b *pathBuilder) EnableNameFieldEscape(enableNameFieldEscape bool) *pathBuilder {
```

Comment:
Why not pass `enableNameFieldEscape bool` to `NewPathBuilder` and eliminate this function?

Reply:
I thought it was not a required value for the pathBuilder, so I didn't put it into the construct function.

Reply:
> I thought it was not a required value for the pathBuilder, so I didn't put it into the construct function.

Looks like you are always calling EnableNameFieldEscape so seems it is required.

`common.NewPathBuilder().EnableNameFieldEscape`

Reply:
Yes, I am always calling the function,  but we will remove it in EdgeX 4.0, so would it be optional in the future?

Reply:
It will be removed, so it will not exist for being optional. Also the clean-up for 4.0 will be easier (slightly) if it is a parameter to the NewPathBuilder. ;-) This not a big deal, so approving PR as is since device services are failing again without it.